### PR TITLE
[Feature][plugins] Add plugin-flink-sink-doris 

### PR DESCRIPTION
**Pull request by JNSimba**
## Purpose of this pull request

Add plugin-flink-sink-doris 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisOutputFormat.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        if (flushException != null) {
+            throw new RuntimeException("Writing records to streamload failed.", flushException);
+        }
+    }
+
+    @Override
+    public synchronized void writeRecord(T row) throws IOException {
+        checkFlushException();
+        addBatch(row);
+        if (batchSize > 0 && batch.size() >= batchSize) {
```
Comment (morningman) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisOutputFormat.java:146]: I think we should make sure that `batchSize` is larger than 0 at begining.

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            maxRetries = config.getInt("doris_sink_max_retries");
+        }
+
+        String producerPrefix = "doris_sink_properties.";
+        PropertiesUtil.setProperties(config, streamLoadProp, producerPrefix, false);
+    }
+
+    @Override
+    public DataSink<Row> outputBatch(FlinkEnvironment env, DataSet<Row> dataSet) {
+        batchIntervalMs = 0;
```
Comment (morningman) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:89]: Why set it to zero here?
And I think `try flush` every 1ms is unnecessary. how about 1 second or maybe longer?

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            BufferedReader br = new BufferedReader(new InputStreamReader(stream));
+            StringBuilder response = new StringBuilder();
+            String line;
+            while ((line = br.readLine()) != null) {
+                response.append(line);
+            }
+            return new LoadResponse(status, respMsg, response.toString());
+
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (yx91490) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java:152]: useless code

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            String err = "failed to stream load data with label:" + label;
+            LOG.warn(err, e);
+            return new LoadResponse(-1, e.getMessage(), err);
+        } finally {
+            if (feConn != null) {
+                feConn.disconnect();
```
Comment (yx91490) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java:158]: what if it throws exception?

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        BatchTableEnvironment tableEnvironment = env.getBatchTableEnvironment();
+        Table table = tableEnvironment.fromDataSet(dataSet);
+        String[] fieldNames = table.getSchema().getFieldNames();
+
+        DorisStreamLoad dorisStreamLoad = new DorisStreamLoad(fenodes, dbName, tableName, username, password, streamLoadProp);
+        return dataSet.output(new DorisOutputFormat(dorisStreamLoad, fieldNames, batchSize, batchIntervalMs, maxRetries));
```
Comment (yx91490) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:95]: new DorisOutputFormat<>

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Override
+    public DataStreamSink<Row> outputStream(FlinkEnvironment env, DataStream<Row> dataStream) {
+        StreamTableEnvironment tableEnvironment = env.getStreamTableEnvironment();
+        Table table = tableEnvironment.fromDataStream(dataStream);
+        String[] fieldNames = table.getSchema().getFieldNames();
+
+        DorisStreamLoad dorisStreamLoad = new DorisStreamLoad(fenodes, dbName, tableName, username, password, streamLoadProp);
+        dataStream.addSink(new DorisSinkFunction<>(new DorisOutputFormat(dorisStreamLoad, fieldNames, batchSize, batchIntervalMs, maxRetries)));
```
Comment (yx91490) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:105]: new DorisOutputFormat<>

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        password = config.getString("password");
+        dbName = config.getString("db_name");
+        if (config.hasPath("doris_sink_batch_size")) {
+            batchSize = config.getInt("doris_sink_batch_size");
+        }
+        if (config.hasPath("doris_sink_interval")) {
+            batchIntervalMs = config.getInt("doris_sink_interval");
+        }
+        if (config.hasPath("doris_sink_max_retries")) {
+            maxRetries = config.getInt("doris_sink_max_retries");
```
Comment (yx91490) [inline seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:80]: valid check, same the above

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java (hunk 4 above)
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java:158]: Thanks, I will modify it

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java (hunk 5 above)
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:95]: Thanks, i am going to modify it

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(DorisStreamLoad.class);
+    private static final List<String> DORIS_SUCCESS_STATUS = new ArrayList<>(Arrays.asList("Success", "Publish Timeout"));
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static String LOAD_URL_PATTERN = "http://%s/api/%s/%s/_stream_load?";
+
+    private String hostPort;
+    private String db;
+    private String tbl;
+    private String user;
+    private String passwd;
```
[the remark this answers is not in the document]
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java:57]: Thanks, I will remove

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+import java.util.Properties;
+import java.util.UUID;
+
+/**
+ * doris streamLoad
+ */
+public class DorisStreamLoad implements Serializable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DorisStreamLoad.class);
+    private static final List<String> DORIS_SUCCESS_STATUS = new ArrayList<>(Arrays.asList("Success", "Publish Timeout"));
```
[the remark this answers is not in the document]
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java:49]: Thanks, i will modify

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisOutputFormat.java (hunk 1 above)
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisOutputFormat.java:146]: Good idea, i will modify

File: seatunnel-connectors/plugin-flink-sink-doris/pom.xml [hunk 10]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (morningman) [reply seatunnel-connectors/plugin-flink-sink-doris/pom.xml:1]: Don't forget this

[the remark this answers is not in the document]
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/pom.xml:1]: thanks, already add license header

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java (hunk 3 above)
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisStreamLoad.java]: already remove

File: seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java (hunk 7 above)
Comment (JNSimba) [reply seatunnel-connectors/plugin-flink-sink-doris/src/main/java/io/github/interestinglab/waterdrop/flink/sink/DorisSink.java:83]: already add check

**Review by CalvinKirs — APPROVED** [review]
LGTM, I will resolve the conflict later, and the naming format needs to be unified.

**Comment by CalvinKirs** [comment]
I merged this PR
But there is a question. Can the relevant configurations of these two PRs(#816 ) be unified? @xtr1993

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**